src/modules/previewpane/powerpreview/settings.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+#include <string>
+#include "resource.h"
+#include <settings_objects.h>
+
+namespace PowerPreviewSettings
+{
+	// PowerToy Winodws Explore File Preview Settings.
+	class FileExplorerPreviewSettings{
+	protected:
```

Comment:
With regards to having the class members private, I thought about it and used the following to guide my judgement:

**private:**
The type or member can only be accessed by code in the same class or struct.

**protected:**
The type or member can only be accessed by code in the same class or struct, or in a derived class.

I wanted to use these variables in derived classes so I went with protected since it still allows me to have the members only accessible to the base and derived classes only.

Reply:
resolved in this commit:
https://github.com/microsoft/PowerToys/pull/1075/commits/f53c46c76327228260ef68eb9ef6e126d5e36c55

Reply:
I was thinking about the second point (b.) All the methods are used outside the class and its derivatives so I had them public.

Reply:
initiliaze m_isPreviewEnabled in the base class:
https://github.com/microsoft/PowerToys/pull/1075/commits/564d677ec40c33e07cb9f713f76c654c14acda98

Reply:
@laviusmotileng-ms:  Thanks for the updates.  I just wanted to provide a little context on my opinion on protected vs private.   Often inside a base class you want to enforce a class 'invariant', but that enforcement might not be known at the time of originally authoring the class.  For example perhaps I have a member variable `double m_velocity`.      If at a later time I decide that in my system I need to enforce that that velocity is between 0.0 and 10.0, I would have to change every class that sets the velocity explicitly.    For this reason I rarely expose member variables as protected, although I might make methods that access or set variables protected if its not something that should be done from code outside the class.

Reply:
Oh, I get what you mean. I will use the getters and settters to update the attributes in derived classes. Thanks.

Reply:
I resolved this here: https://github.com/microsoft/PowerToys/pull/1075/commits/796a90a3086796518e9ab396a6d281323d80e900